### DISCUSSION TITLE
Phd Thesis Covers - UMINHO/ee

### DISCUSSION THREAD

**Discussion by Marcelo-Alves7** [joaomlourenco/novathesis#361] [Q&A] [answered]
Hello professor and community,

I am finishing my PhD thesis at the Uminho School of Engineering and my supervisor called my attention to correctly complete the cover (back + spine + front together, in book style - send photo) in order to respect the rules of the covers.

I thought that changing the docstatus to final would change the covers, but no. 
Can you help me with this issue?

Thank you!!
![capa_UM_ee](https://user-images.githubusercontent.com/98588105/192793317-cbff5b11-6088-4312-bd38-72472b87fa52.png)

**Comment by b-pereira** [comment]
As far as I know, you can't with the current template (check #254). What you can do is the following:

1. Edit the custom "back cover + spine + cover" as you show above, modifying the AI files from the UMINHO's channels (~~https://alunos.uminho.pt/PT/estudantes/Formataes/Logos.zip~~ https://alunos.uminho.pt/PT/estudantes/Formataes/Capas.zip) **with -3mm margin** with Inkscape, for instance.
2. Edit the verse **with the same dimensions** as the "back cover + spine + cover" **with -3mm margin**;
3. Include the package `pdfpages` - `\usepackage{pdfpages}`
4. Find the following line in file `template.tex` and comment it (just check if it has the frontpage after):
![imagem](https://user-images.githubusercontent.com/944511/192802135-1554fb3f-a7f2-46d6-8409-e0a7af71a040.png)
5. Add the `\includepdf` command for the PDFs "back cover + spine + cover" and verse as below in the place of the previous commented line;
```latex
 \setcounter{page}{0}
 \eject \pdfpagewidth=440mm \pdfpageheight=297mm
 \includepdf[fitpaper=true]{backcover_spine_cover.pdf}
 \setcounter{page}{0}
 \includepdf[width=440mm, height=297mm, fitpaper=true]{verse.pdf}
 \eject \pdfpagewidth=210mm \pdfpageheight=297mm
 ```
6. Compile and _shazam!_ **Note that the dimensions in the first `\pdfpagewidth` are considering -3 mm all around the  "back cover + spine + cover"**. The reason is that UMINHO provided AI files with a +3mm all around, which is in issue #254 - the bleeding margin for the trimming at the printing company. Also, I checked that it works without the `\pdfpagewidth` command in this template, but let it be just to guarantee that the page has those dimensions and return to A4 format. **If the front page is an issue you can export it to PDF and include it after the second `\eject` command.** Or if your **latex skills** are up to standard, check this https://github.com/b-pereira/UMINHO-PHD-MSc-Covers/releases/tag/v1.0.0-beta.1 I didn't have the time to fully implements the features for all the "back cover + spine + cover", verse and front page.

**Reply by Marcelo-Alves7** [reply]
Thanks for the quick response.

1- The link you sent concerns the logos, I think it's wrong. It should be the covers - https://alunos.uminho.pt/PT/estudantes/Formataes/Capas.zip
Am I right?

2- As for modifying the channel's AI files, do I need to save the pdf already modified (-3mm margin with Inkscape) or will it be just a case when I include the command you pre-defined?

3- where can I find the verse to edit in the same way as the "back cover + spine + cover" with -3mm margin? 
The download only contains the "Capa-Doutoramento", "Capa Mestrado" and "FOLHA-ROSTO-EE".

4- Do I need to include the pdfs in a specific folder?

Thank you once again.

**Comment by b-pereira** [comment, marked as answer]
You're welcome!

1. Yes, you're right. The URL is wrong and that's the correct URL.

2. The idea is to not include this template's covers. So, you **must** edit the AI file and export it to PDF, and choose the name you want for the file. I used `backcover_spine_cover.pdf` and `verse.pdf` to illustrate the example. Also, it's on export that you define a -3mm border.

3. If you have Adobe Illustrator, it's the second page of the CAPA-* file (Mestrado or Doutoramento) - it's included in those AIs. However, I couldn't open and view the second page with Inkscape. Notwithstanding, you can just **crop** from the manual or the logo and export the image to PDF. Note the `fitpaper=true` parameter. You can just have a 1mmx1mm square PDF that will fit the page size - it's just color so, you won't have any issue. You can, alternatively, use https://github.com/b-pereira/UMINHO-PHD-MSc-Covers/releases/tag/v1.0.0-beta.1 and follow the instructions. You can compile covers and verses and then the front page separately (just choose the file as you do with `template.tex`). You have to input the values directly, however (put the title 3 times - cover, spine, and front page -, put the author name 2 times (cover and front page),- put the simplified author name on the spine, as well "UMinho | 2022". I can't remember if I added the automatic date, but if I didn't you have to put it twice on the cover and the front page. If I had the time to program that correctly ... it wouldn't be that cumbersome. Hence the *pre-release*.

4. As I mentioned in 2. I included an illustrative name file, **without folders**. So, you can just use the project's root path as it is, or you can add a folder and then  `myFolder/backcover_spine_cover.pdf` and `myFolder/verse.pdf`. Note that this is a quick fix that goes beyond this template's functionality.

**Don't forget to mark this as answered!**

**Reply by Marcelo-Alves7** [reply]
Thanks for the prompt reply once again.

I'd just like to ask you two more quick questions that I'm a little confused here.

1- I'm trying to use Inkscape (which I've never used) and when I upload the AI ​​file due, I seem to be able to see both pages (Cover + Verse). Then what I have to do is export page by page, where in each export I reduce -3mm margin. That is, I select cover_spine_cover and export and where do I reduce the -3mm? Where can I customize this measurement? Do I have to change the units of measurement? Sorry for these doubts.
Then do something similar for the back page (red part of the image)
![1_teste_capa](https://user-images.githubusercontent.com/98588105/192855786-456bbd5e-b37e-4ff2-926c-fb8d8ae45a78.png)

2- As I have doubts, I tried the second approach, https://github.com/b-pereira/UMINHO-PHD-MSc-Covers/releases/tag/v1.0.0-beta.1, but the instructions do not reveal where to include the commands described in the general NOVAThesis template? Will it be in template.tex or something more specific? It looks like you built something from root with a new main.text.

Sorry for the trouble, but since I'm a little worried, I can't even follow up on things.
Thanks.

**Reply by b-pereira** [reply]
1. In the image that you provided, on the right side, below *Image Size* you can encounter the image **selected**. Left of it, you see **Export selected only**. Never saw that, maybe because I didn't enlarge the window enough to see them both. You also can see **Page 1**.
2. https://github.com/b-pereira/UMINHO-PHD-MSc-Covers/releases/tag/v1.0.0-beta.1 has nothing to do with NOVAThesis. **DON'T ADD IT TO THE NOVATHESIS TEMPLATE!** **CREATE A NEW LATEX PROJECT AND EXPORT THE PDFS.** It's *standalone*. You have to fill the fields **manually**, as I mentioned before. The example that I gave in the *pre-release* README has the following code snippet:
```latex
\newsavebox\AuthorBox
\sbox\AuthorBox{%
\begin{minipage}[b]{134mm}
\color{white}
\fontsize{17pt}{20.4pt}\selectfont{\sffamily\color{white}{Nome do Autor}}
\vspace{\baselineskip}\\
\fontsize{17pt}{20.4pt}\selectfont{\sffamily\color{white}{\textbf{Título Título Título Título Título Título\\Título Título Título Título Título\\Título Título Título Título}}}
\end{minipage}
}%
```
In there you can see "Nome do Autor" and "Título Título Título Título Título Título\\\Título Título Título Título Título\\\Título Título Título Título" that has breaklines. So, you have to change that directly in the code. Check the example below:

```latex
\newsavebox\AuthorBox
\sbox\AuthorBox{%
\begin{minipage}[b]{134mm}
\color{white}
\fontsize{17pt}{20.4pt}\selectfont{\sffamily\color{white}{FULANO SICRANO}}
\vspace{\baselineskip}\\
\fontsize{17pt}{20.4pt}\selectfont{\sffamily\color{white}{\textbf{A VERY VERY VERY VERY VERY BIG\\AND  VERY VERY VERY IMPRESSIVE\\THESIS TITLE }}}
\end{minipage}
}%
```
You have a `\SpineBox` a `\DateBox` and so on. And it's always the same: Find the field and fill it **manually**. Well, maybe I will finally do a usable version, that has automation. There aren't many resources on TeX Programming - it's OOOLD xD.

**Comment by Marcelo-Alves7** [comment]
Okay, I think I got it and thanks for all the attention.

I will opt for the second approach and use the latex skills already programmed and fill in the fields manually to generate and export the PDFs.
From what I've seen, the generated pdf only generates the "cover_spine_cover.pdf". What do I have to do to get the verse.pdf (the second red page)?

After that, I still have to separate the pdfs to include them in NOVAThesis as you mentioned in your first answer:
 \setcounter{page}{0}
 \eject \pdfpagewidth=440mm \pdfpageheight=297mm
 \includepdf[fitpaper=true]{backcover_spine_cover.pdf}
 \setcounter{page}{0}
 \includepdf[width=440mm, height=297mm, fitpaper=true]{verse.pdf}
 \eject \pdfpagewidth=210mm \pdfpageheight=297mm
 
  Regarding \ntprint{backcover}, não será também necessária alguma alteração?
  If I change the NOVAThesis template to the final version, the following appears:
![image](https://user-images.githubusercontent.com/98588105/192873360-4fe242fa-1364-452d-8ab4-2ecb88a74f7e.png)
Ends with a blank A4 page, split in half. will it be like that?
 
  Thanks and sorry for the number of questions.

**Reply by b-pereira** [reply]
Ya, comment that line. 

```latex
%\ntprint{backcover}
```

Also, find the spine command in `template.tex` and change it to no. What "Ends with a blank A4 page, split in half" it's supposed to be the spine.

```latex
%\ntsetup{spine=trim}
\ntsetup{spine=no}
```

btw, don't mix English with Portuguese. Strict to the maintainer's specified rules. But let it be.
> Regarding \ntprint{backcover}, não será também necessária alguma alteração?

**Comment by Marcelo-Alves7** [comment]
Got it, and sorry, I'm already at a stage where I mix everything up.

Finally, it didn't answer how can I generate the verse according to your code (project).
From what I've seen, the generated pdf only generates the "backcover_spine_cover.pdf". 
What do I have to do to get the verse.pdf (the red PANTONE 1807 C page)?

Thanks
